### PR TITLE
build(schemas): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

### CODE REVIEW

File: experimental/PropertyDDS/examples/schemas/package.json [hunk 1]
```diff
@@ -11,29 +11,44 @@
...
 	"types": "dist/index.d.ts",
 	"scripts": {
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
+		"build:esnext": "tsc --project ./tsconfig.json",
+		"check:are-the-types-wrong": "attw --pack . --entrypoints .",
 		"check:biome": "biome check .",
 		"check:prettier": "prettier --check . --cache --ignore-path ../../../../.prettierignore",
-		"clean": "rimraf --glob dist \"**/*.tsbuildinfo\" \"**/*.build.log\"",
+		"clean": "rimraf --glob dist \"**/*.tsbuildinfo\" \"**/*.build.log\" lib",
```
Comment (jason-ha) [inline experimental/PropertyDDS/examples/schemas/package.json:36]: sort lib to be just after dist ... if we actually need to dual build this. Can we just leave it CJS?

File: experimental/PropertyDDS/examples/schemas/tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {
-	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"extends": "../../../../common/build/build-common/tsconfig.node16.json",
 	"compilerOptions": {
-		"composite": true,
 		"rootDir": "./src",
-		"outDir": "./dist",
+		"outDir": "./lib",
 		"baseUrl": ".",
```
Comment (jason-ha) [inline experimental/PropertyDDS/examples/schemas/tsconfig.json:6]: What is baseUrl doing?

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: experimental/PropertyDDS/examples/schemas/tsconfig.json (hunk 2 above)
Comment (RishhiB) [reply experimental/PropertyDDS/examples/schemas/tsconfig.json:6]: schemas src code and tests already use relative paths, so it should be fine to remove this line. I will remove the line and test.

File: experimental/PropertyDDS/examples/schemas/package.json (hunk 1 above)
Comment (RishhiB) [reply experimental/PropertyDDS/examples/schemas/package.json:36]: @jason-ha, How do I know if I should leave it CJS? Also did you want to change the script to;
```suggestion
		"clean": "rimraf --glob dist lib \"**/*.tsbuildinfo\" \"**/*.build.log\"",
```

Comment (tylerbutler) [reply experimental/PropertyDDS/examples/schemas/package.json:36]: Yes, I think moving lib to be after dist in the command is what he meant.

**Comment by sonalideshpandemsft** [comment]
closing in favor of https://github.com/microsoft/FluidFramework/pull/20311

**Pull request closed without merging**